docs/proposals/multi-platform.md

```diff
@@ -0,0 +1,477 @@
+
+The crucial thing here is that the components have to expose their platform, and initially we've solved it by exposing the labels `beta.kubernetes.io/{os,arch}` on every node. This way an user may run binaries for different platforms on a multi-platform cluster, but still it requires manual work.
+
+Also, [the apiserver now exposes](https://github.com/kubernetes/kubernetes/pull/19905) it's platform at `GET /version`. But note that the value exposed at `/version` only is the apiserver's platform; there might be kubelets of various other platforms.
+
+### Vendored packages
+
+Kubernetes might depend on packages that aren't coded in a cross-platform manner.
+Here are two common problems:
+ - Including constants combined with build tags
+```go
+//+ build linux,amd64
+const AnAmd64OnlyConstant = 123
+```
+ - Relying on platform-specific syscalls (e.g. `syscall.Dup2`)
+
+If this happens, we have to send a PR that fixes it in the vendored repository. Only after that, we may add or update the dependency.
+
+## Building and releasing
+
+The major part of this work affects the test and release infrastructure system. The builds are running inside a docker container; `kube-cross`, where all `linux/amd64` build dependencies are installed in order to get reproducible builds. This means `build/run.sh` commands only work on `linux/amd64`, but the main compilation script `hack/build-go.sh` should work on all platforms given that dependencies like `go` and `gcc` are installed.
+
+The released binaries are uploaded to `https://storage.googleapis.com/kubernetes-release/release/${version}/bin/${os}/${arch}/${binary}`
+
+### Image naming
+
+This has been debated quite a lot about; how we should name non-amd64 docker images that are pushed to `gcr.io`. See [#23059](https://github.com/kubernetes/kubernetes/pull/23059) and [#23009](https://github.com/kubernetes/kubernetes/pull/23009). The conclusion is that until docker incorporates the manifest list fully, we will use the `gcr.io/google_containers/${binary}-${arch}:${version}` naming.
+
+This means that the "older" naming `gcr.io/google_containers/${binary}:${version}` is deprecated, _but only for those images that are compiled for multiple platforms._
+
+### Client binaries
+
+Client binaries, i. e. only kubectl at the moment, are built for more platforms than the server binaries are.
+If you want to check out an up-to-date list of them, [see here](https://github.com/kubernetes/kubernetes/blob/master/hack/lib/golang.sh).
+
+kubectl is a static binary with no C code, so it's trivial to cross-compile. If there's interest in adding a client platform, it should be as easy as appending the list.
+
+### Running a CI
+
+To ensure all functionality really is working on an other platform, the community should be able to setup a CI. To be able to do that, all the test-specific images have to be ported to multiple architectures. The test code also have to choose images based on the apiserver's platform.
```

Comment:
I'm fundamentally in agreement here... but I would like to start by making our CI reproducible on one platform... then get started on other platforms.

Trying to run testing right now is pretty byzantine. So I'm hesitant to spend much time making our current testing images/scripts multi-platform when they are likely to change so dramatically in the next few months.

The linked test-infra issues are highly relevant, as well as https://github.com/kubernetes/test-infra/pull/234